extension/storage/local/filestorage/extension.go

```diff
@@ -0,0 +1,106 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filestorage
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"go.etcd.io/bbolt"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/storage"
+)
+
+type localFileStorage struct {
+	directory string
+	timeout   time.Duration
+	logger    *zap.Logger
+	clients   []*fileStorageClient
+}
+
+// Ensure this storage extension implements the appropriate interface
+var _ storage.Extension = (*localFileStorage)(nil)
+
+func newLocalFileStorage(logger *zap.Logger, config *Config) (component.Extension, error) {
+	info, err := os.Stat(config.Directory)
+	if (err != nil && os.IsNotExist(err)) || !info.IsDir() {
+		return nil, fmt.Errorf("directory must exist: %v", err)
+	}
+
+	testFile := filepath.Join(config.Directory, "test.tmp")
+	options := &bbolt.Options{Timeout: time.Second}
+	db, err := bbolt.Open(testFile, 0666, options)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Why do we need to test it?

Reply:
We don't necessarily need to. In fact, the same is done at client creation.

I think I'll remove this. However, one possible reason to keep it is that the error is then associated with the extension, rather than with the component using the extension, which may be more helpful to the user.